### ISSUE TITLE
Trouble Verifying Custom SuperTokens on Goerli

### ISSUE THREAD

**Issue by mikeghen** [superfluid-finance/custom-supertokens#22]
Vijay: seems like Etherscan have stopped supporting the automatic proxy verification, you might have to click "is this a proxy?" and letting it match the default one for super token wrappers for etherscan interactions for a while...

https://discord.com/channels/752490247643725875/752493319657095168/1089949208464867418

Could this be why I'm suddenly having an issue verifying supertokens on Goerli? 

```shell
$ npx truffle run --network goerli verify BurnMintSuperToken@0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a --custom-proxy BurnMintSuperToken

Verifying contracts on etherscan
   Verifying BurnMintSuperToken@0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a
   Verifying custom proxy contract BurnMintSuperToken at 0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a
   Verifying proxy implementation BurnMintSuperToken at 0xdcde53f3b5e275109ff105f4dcab6925fc62791a
   Invalid constructor arguments provided. Please verify that they are in ABI-encoded format
   Failed to verify 1 contract(s): BurnMintSuperToken@0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a
```

Or is this something related to how constructor args might have been done on token deployment. I think I might have done something wrong on deployment.

**Comment by d10r** [comment]
I suspect that it verified the proxy at the first attempt, but failed with the logic contract and linking.
Our issue here is that there seems to be no way with truffle-plugin-verify to verify only the proxy contract, not the implementation. It tries to be smart and verify the logic the proxy points to. See https://github.com/rkalis/truffle-plugin-verify/issues/178

In this project, we're kind of cheating, because the code of the implementation contract is outside of this project. A more correct cmdline for verification would actually be:
```
npx truffle run --network goerli verify SuperToken@0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a --custom-proxy BurnMintSuperToken
```

However we don't have a`SuperToken` artifact here. 
It probably wouldn't be too difficult to get it (since we have the ethereum-contracts npm package), but even then its bytecode would likely often not match the one currently deployed, because the package version may not match the version of the currently deployed contracts etc.

So, it will fail when trying to verify the implementation contract, but succeed with verifying the proxy contract, which is what we need.
I think it didn't fail in the past if the implementation contract was already verified, but am not sure. Maybe that changed, or maybe it's related to changes in the canonical SuperToken logic contract (it's nomore deployed by the SuperTokenFactory contract in 1.5.2+).

Long story short: I don't currently see an obvious way to properly fix this without making it more fragile.
But there's a pretty simple way to work around it: ignore the error message and manually do the "linking" between proxy and logic in the explorer.
I updated the README accordingly in this PR: https://github.com/superfluid-finance/custom-supertokens/pull/23/files

I didn't do it for 0xA862f128b3ffE9719948869cc1C7CBC21860Cd9a, so you can try it yourself.
Let me know if that's not working as expected or if you have any more questions or suggestions.
And thx for the report!
